src/richie/apps/courses/templates/courses/cms/category_detail.html

```diff
@@ -24,6 +28,7 @@ <h1 class="category-detail__title">{% render_model request.current_page "title"
 </div>
 
 
+{% if category.get_blogposts %}
```

Comment:
this will do the query twice... once here and once in the for loop below. I think it's better to wrap everything in a {% with blogposts = category.get_blogposts %} clause.